app/controllers/devise/devise_authy_controller.rb

```diff
@@ -82,33 +82,23 @@ def POST_enable_authy
     end
   end
 
-  # Disable 2FA
+  # Disables MFA and deletes all MFA config for the resource across all factors.
   def POST_disable_authy
-    authy_id = resource.authy_id
-    resource.assign_attributes(:authy_enabled => false, :authy_id => nil)
-    resource.save(:validate => false)
-
-    other_resource = resource.class.find_by(:authy_id => authy_id)
-    if other_resource
-      # If another resource has the same authy_id, do not delete the user from
-      # the API.
+    mfa_config = resource.mfa_config
+
+    begin
+      delete_entity(mfa_config.verify_identity)
+      mfa_config.delete
+      resource.assign_attributes(authy_enabled: false, authy_id: nil)
+      resource.save(validate: false)
```

Comment:
@BintLopez I wrapped lines 91-93 in a transaction block and pushed. Wrapping delete_entity doesn't really help us because it's a deletion on Twilio itself and we don't have a way to undo it if saving the db fails.